### PR TITLE
docs: replace react-color with color-picker

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
#42333 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     replace react-color with color-picker       |
| 🇨🇳 Chinese |    将 react-color 替换为 color-picker       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at c663409</samp>

Refactored various components and demos to use the `antd` ColorPicker component and the `antd` color util functions, instead of the `react-color` and `TinyColor` libraries. This improved the UI consistency, code quality, and dependency management. Fixed some minor bugs and code style issues along the way.

### <samp>🤖 Generated by Copilot at c663409</samp>

*  Replace `react-color` SketchPicker component with `antd` ColorPicker component and use `antd` color util functions and `Color` type instead of `TinyColor` library and `RGBColor` type in various files and components ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-bb71d3e7437a45c3978afff7a00abefa4e0c15a401677f1a732c773f7027d36eL1-R8),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-bb71d3e7437a45c3978afff7a00abefa4e0c15a401677f1a732c773f7027d36eL55-R83),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-5bbc7ab290ac59ef4fc2d700df8c3db8b4ca887c53c3cf964dee496835d94488L1-R4),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-5bbc7ab290ac59ef4fc2d700df8c3db8b4ca887c53c3cf964dee496835d94488L12-R14),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-5bbc7ab290ac59ef4fc2d700df8c3db8b4ca887c53c3cf964dee496835d94488L23-R40),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-f614cf350953f24221b4e3d9cf43608ae3346bdea3970ee6b26752037d4f50deL1-R3),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-f614cf350953f24221b4e3d9cf43608ae3346bdea3970ee6b26752037d4f50deL16-R20),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-f614cf350953f24221b4e3d9cf43608ae3346bdea3970ee6b26752037d4f50deL34-R35),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-f614cf350953f24221b4e3d9cf43608ae3346bdea3970ee6b26752037d4f50deL53-R58),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-f614cf350953f24221b4e3d9cf43608ae3346bdea3970ee6b26752037d4f50deL75-R73),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-a19cb548fa41fcf5933a871bf0bfffffe96c649e60dcab61be32b7f97f8a3f8dL1-R3),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-a19cb548fa41fcf5933a871bf0bfffffe96c649e60dcab61be32b7f97f8a3f8dL28-R28),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-a19cb548fa41fcf5933a871bf0bfffffe96c649e60dcab61be32b7f97f8a3f8dL35-R47),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7L1-R25),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7L126-R130))
* Rename `DebouncedColorPanel` component to `DebouncedColorPicker` and change its props to match `antd` ColorPicker props in `ColorPicker.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-bb71d3e7437a45c3978afff7a00abefa4e0c15a401677f1a732c773f7027d36eL41-R40))
* Move `DEFAULT_COLOR` constant from `colorUtil.ts` file to `ColorPicker.tsx` file, as it is only used in the latter ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-b496a4a3b3a72cf12f1ec43f6c4446445b8422cabed134b35633bb6cc46d6419L1-R1))
* Destructure `themeType`, `colorPrimary`, and `borderRadius` properties from `themeData` object in `index.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL21-R36))
* Add `colorPrimaryValue` variable to ensure a valid hex color string is used in various functions and components in `index.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL224-R230),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL239-R240),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL245-R246),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL312-R317),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL319-R324),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL331-R343),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedR357),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL510-R522),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL558-R576))
* Change `getTitleColor` function to use `toHsb` method of `Color` type instead of `toHsl` method of `TinyColor` type in `index.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL239-R240))
* Move `isLight` variable to the top of the component body in `index.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL283-R291))
* Change `logoColor` variable to use `generateColor` function and `toHsb` and `toHexString` methods of `Color` type instead of `TinyColor` constructor and `toHsl` and `toHexString` methods of `TinyColor` type in `index.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL331-R343))
* Change `colorNode` to use `DebouncedColorPicker` component instead of `Popover` and `DebouncedColorPanel` components in `ColorPicker.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-bb71d3e7437a45c3978afff7a00abefa4e0c15a401677f1a732c773f7027d36eL129-R143))
* Change `value` prop of `Input` component to use `toHexString` method of `Color` type if the value is not a string in `ColorPicker.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-bb71d3e7437a45c3978afff7a00abefa4e0c15a401677f1a732c773f7027d36eL93-R107))
* Add `width` property to the input style in `input.ts` file to fix a bug that caused the input to shrink when the color picker was opened ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-76a3207c1405e5d3615abf7040b26c6e5366e5d9b863ba2f9a78194d727fe2d7R45))
* Change `watermark` config to use `color` property instead of `font.color` property and to use `typeof` operator and `toRgbString` method of `Color` type to ensure a valid rgb color string is used in `custom.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7L61-R34))
* Change double quotes to single quotes for consistency with the rest of the code style in various files ([link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-bb71d3e7437a45c3978afff7a00abefa4e0c15a401677f1a732c773f7027d36eL23-R22),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-bb71d3e7437a45c3978afff7a00abefa4e0c15a401677f1a732c773f7027d36eL102-R114),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-bb71d3e7437a45c3978afff7a00abefa4e0c15a401677f1a732c773f7027d36eL119-R131),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL402-R408),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL409-R415),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL415-R421),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL430-R438),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL454-R461),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL467-R486),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL534-R541),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedL543-R550),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7L107-R81),[link](https://github.com/ant-design/ant-design/pull/42350/files?diff=unified&w=0#diff-26c21a1646213a2bfb681dfe72825f676943f442103ea720c4765ab70a71f0c7L120-R93))

### CODE REVIEW

File: components/color-picker/style/input.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ const genInputStyle: GenerateStyle<ColorPickerToken, CSSObject> = (token) => {
 
       [`${componentCls}-format-select${antCls}-select`]: {
         marginInlineEnd: marginXS,
+        width: 'auto',
```
Comment (yoyo837) [inline components/color-picker/style/input.ts:45]: 这个改动是站点之外的噢，是直接改组件，是个修复吗？ 如果是， 是不是另起fix PR好一点？

Comment (RedJue) [reply components/color-picker/style/input.ts:45]: 可以 那我一会重新起一个pr

**Comment by zombieJ** [comment]
样式不对，预设没有文字这个看起来很奇怪：

截屏2023-05-15 14 07 54

**Comment by RedJue** [comment]
> 样式不对，预设没有文字这个看起来很奇怪：
> 
> 截屏2023-05-15 14 07 54

Ref: #42356

File: .dumi/pages/index/components/Theme/ColorPicker.tsx [hunk 2]
```diff
@@ -128,7 +128,7 @@ export default function ThemeColorPicker({ value, onChange }: RadiusPickerProps)
               }}
             >
               <input
-                type="radio"
+                type='radio'
```
Comment (afc163) [inline .dumi/pages/index/components/Theme/ColorPicker.tsx:131]: 不应该改这些。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**